crates/harness/src/monolith.rs

```diff
@@ -105,20 +116,20 @@ impl Monolith {
                                 match msg {
                                     Ok(msg) => {
                                         println!("monolith: incoming msg: {}", msg);
-                                        let mut state = state.lock().unwrap();
-                                        state.received_raw.push(msg.clone());
-                                        // TODO: there's a better way to generalize this
-                                        if let Message::Text(m) = msg {
-                                            let msg: MsgB2M = serde_json::from_str(&m).unwrap();
-                                            match msg {
-                                                MsgB2M::Join(join) => {
-                                                    state.clients.insert(join.client);
-                                                },
-                                                MsgB2M::Leave(leave) => {
-                                                    state.clients.remove(&leave.client);
-                                                },
-                                                _ => {},
-                                            }
+                                        let to_send = {
+                                            let mut state = state.lock().unwrap();
```

Comment:
This only occurred to me now but I see .lock().unwrap() a lot. Why?

Reply:
`state` is a `Arc<Mutex<T>>`. The `Arc` pointer gets implicitly dereferenced, but in order to safely access the contents of a Mutex, you have to lock it first. The result of that is you get a `MutexGuard` that automatically releases the lock when it's dropped.